### PR TITLE
fix test flake in TestLegacyServiceAccountTokenTracking

### LINKED ISSUE #113821: [Flake] TestLegacyServiceAccountTokenTracking flakes on number of warnings received
### Which jobs are flaking?

pull-kubernetes-integration

### Which tests are flaking?

TestLegacyServiceAccountTokenTracking (in top 10 list of flakes - http://storage.googleapis.com/k8s-metrics/flakes-latest.json)

### Since when has it been flaking?

10/25

https://storage.googleapis.com/k8s-triage/index.html?pr=1&test=TestLegacyServiceAccountTokenTracking

### Testgrid link

_No response_

### Reason for failure (if possible)

It looks like the number of warnings received is not consistent:
```
    service_account_test.go:385: Expect 8 warnings, got 6
...
    service_account_test.go:385: Expect 8 warnings, got 7
```

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig auth
/assign @zshihang 
/milestone v1.26
/priority important-soon

we should resolve this flake before test freeze

### PR DESCRIPTION
**Pull request by zshihang**
#### What type of PR is this?

/kind flake

#### What this PR does / why we need it:

i couldn't reproduce it locally. eyeballing the code.

#### Which issue(s) this PR fixes:

Fixes #113821

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/integration/serviceaccount/service_account_test.go [hunk 1]
```diff
@@ -371,18 +371,26 @@ func TestLegacyServiceAccountTokenTracking(t *testing.T) {
...
-			}()
-			doServiceAccountAPIRequests(t, roClient, myns, true, true, false)
+
+			var wg sync.WaitGroup
+			concurrency := 5
+			for i := 0; i < concurrency; i++ {
+				wg.Add(1)
+				go func() {
+					doServiceAccountAPIRequests(t, roClient, myns, true, true, false)
+					wg.Done()
```
Comment (aojea) [inline test/integration/serviceaccount/service_account_test.go:381]:
```suggestion
                                          defer wg.Done()
					doServiceAccountAPIRequests(t, roClient, myns, true, true, false)
```

Comment (zshihang) [reply test/integration/serviceaccount/service_account_test.go:381]: one line to defer? probably not.

File: test/integration/serviceaccount/service_account_test.go [hunk 2]
```diff
@@ -371,18 +371,26 @@ func TestLegacyServiceAccountTokenTracking(t *testing.T) {
...
+			wg.Wait()
 			dateAfter := time.Now().UTC().Format(dateFormat)
 			liveSecret, err := c.CoreV1().Secrets(myns).Get(context.TODO(), test.secretName, metav1.GetOptions{})
 			if err != nil {
 				t.Fatalf("Could not get secret: %v", err)
 			}
 
-			if test.expectWarning && len(wh.warnings) != 8 {
-				t.Fatalf("Expect 8 warnings, got %d", len(wh.warnings))
+			// doServiceAccountAPIRequests has 4 API requests
```
Comment (aojea) [inline test/integration/serviceaccount/service_account_test.go:391]: is not running 5 times?

Comment (zshihang) [reply test/integration/serviceaccount/service_account_test.go:391]: in doServiceAccountAPIRequests method, there are 4 API requests. we expect 4 * 5 = 20 warnings

File: test/integration/serviceaccount/service_account_test.go (hunk 1 above)
Comment (aojea) [reply test/integration/serviceaccount/service_account_test.go:381]: is the common pattern, if in the future you add more lines, this is always covered ... but is only nit, don't worry

File: test/integration/serviceaccount/service_account_test.go (hunk 2 above)
Comment (aojea) [reply test/integration/serviceaccount/service_account_test.go:391]: oh, right, concurrency is 5, need some sleep , sorry

**Comment by aojea** [comment]
/lgtm

Thanks

```
2m25s: 493 runs so far, 0 failures
2m30s: 506 runs so far, 0 failures
2m35s: 540 runs so far, 0 failures
2m40s: 544 runs so far, 0 failures
2m45s: 578 runs so far, 0 failures
2m50s: 591 runs so far, 0 failures
2m55s: 599 runs so far, 0 failures
3m0s: 636 runs so far, 0 failures
^C
```

File: test/integration/serviceaccount/service_account_test.go [hunk 3]
```diff
@@ -371,18 +371,26 @@ func TestLegacyServiceAccountTokenTracking(t *testing.T) {
...
+			var wg sync.WaitGroup
+			concurrency := 5
+			for i := 0; i < concurrency; i++ {
+				wg.Add(1)
+				go func() {
+					doServiceAccountAPIRequests(t, roClient, myns, true, true, false)
+					wg.Done()
+				}()
+			}
+			wg.Wait()
```
Comment (liggitt) [inline test/integration/serviceaccount/service_account_test.go:384]: this was the flaky bit? we weren't waiting for the async doServiceAccountAPIRequests invocation to finish before?

Comment (zshihang) [reply test/integration/serviceaccount/service_account_test.go:384]: yea, forgot it :(

**Comment by liggitt** [comment]
/lgtm
/approve
/milestone v1.26

adding test-only change to milestone ahead of test freeze to resolve flake

**Pull request merged**